zerver/lib/user_groups.py

```diff
@@ -766,7 +759,19 @@ def set_defaults_for_group_settings(
         else:
             default_group_name = permission_config.default_group_name
 
-        default_group = system_groups_name_dict[default_group_name].usergroup_ptr
+        if default_group_name == "group_creator":
+            if user_group.creator:
+                default_group = UserGroup(
+                    realm=user_group.realm,
+                )
+                default_group.save()
+                UserGroupMembership.objects.create(
+                    user_profile=user_group.creator, user_group=default_group
+                )
```

Comment:
Could this share code with `update_or_create_user_group_for_setting`?

Reply:
That will increase the query count, see https://github.com/zulip/zulip/pull/31596#discussion_r1777269763